src/main/java/com/shipmentEvents/demo/BufferedReaderExample.java

```diff
@@ -2,15 +2,39 @@
 
 import java.io.BufferedReader;
 import java.io.FileReader;
+import java.io.FileWriter;
+import java.util.Scanner;
 
 public class BufferedReaderExample {
     public static void main(String[] args) {
+        writeToFile(promptForContact());
+    }
+
+    private static Contact promptForContact() {
+        Scanner in = new Scanner(System.in);
+
+        //adding a line here testing things
+        System.out.println("Please enter the contact name:");
+        String name = in.nextLine();
+
+        System.out.println("Please enter the contact phone number:");
+        String number = in.nextLine();
+
+        Contact person = new Contact();
+        person.setName(name);
+        person.setPhoneNumber(number);
+
+        return person;
+    }
 
 
-            try {
-                BufferedReader br = new BufferedReader(new FileReader("test.txt"));
-            } catch (Exception e) {
-                System.out.println(e.getMessage());
-            }
+    public static void writeToFile(Contact contact) {
+        try {
+            //BufferedReader br = new BufferedReader(new FileReader("test.txt"));
+            FileWriter myWriter = new FileWriter("contacts.txt");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *myWriter*. Execution paths that do not contain closure statements were detected. Ensure *myWriter* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).